eventsource-client/src/event_parser.rs

```diff
@@ -379,39 +384,59 @@ mod tests {
+        let mut bytes = Vec::from("id: ");
+        let mut invalid = vec![b'\xf0', b'\x28', b'\x8c', b'\xbc'];
+        bytes.append(&mut invalid);
+        bytes.push(b'\n');
+        let mut parser = EventParser::new();
+        assert!(parser.process_bytes(Bytes::from(bytes)).is_err());
+    }
+
     #[test]
     fn test_parse_field_comments() {
-        assert_eq!(parse_field(b":"), field("comment", b""));
+        assert_eq!(parse_field(b":"), field("comment", ""));
         assert_eq!(
             parse_field(b":hello \0 world"),
-            field("comment", b"hello \0 world")
+            field("comment", "hello \0 world")
         );
-        assert_eq!(parse_field(b":event: foo"), field("comment", b"event: foo"));
+        assert_eq!(parse_field(b":event: foo"), field("comment", "event: foo"));
     }
 
     #[test]
     fn test_parse_field_valid() {
-        assert_eq!(parse_field(b"event:foo"), field("event", b"foo"));
-        assert_eq!(parse_field(b"event: foo"), field("event", b"foo"));
-        assert_eq!(parse_field(b"event:  foo"), field("event", b" foo"));
-        assert_eq!(parse_field(b"event:\tfoo"), field("event", b"\tfoo"));
-        assert_eq!(parse_field(b"event: foo "), field("event", b"foo "));
+        assert_eq!(parse_field(b"event:foo"), field("event", "foo"));
+        assert_eq!(parse_field(b"event: foo"), field("event", "foo"));
+        assert_eq!(parse_field(b"event:  foo"), field("event", " foo"));
+        assert_eq!(parse_field(b"event:\tfoo"), field("event", "\tfoo"));
+        assert_eq!(parse_field(b"event: foo "), field("event", "foo "));
 
-        assert_eq!(parse_field(b"disconnect:"), field("disconnect", b""));
-        assert_eq!(parse_field(b"disconnect: "), field("disconnect", b""));
-        assert_eq!(parse_field(b"disconnect:  "), field("disconnect", b" "));
-        assert_eq!(parse_field(b"disconnect:\t"), field("disconnect", b"\t"));
+        assert_eq!(parse_field(b"disconnect:"), field("disconnect", ""));
+        assert_eq!(parse_field(b"disconnect: "), field("disconnect", ""));
+        assert_eq!(parse_field(b"disconnect:  "), field("disconnect", " "));
+        assert_eq!(parse_field(b"disconnect:\t"), field("disconnect", "\t"));
 
-        assert_eq!(parse_field(b"disconnect"), field("disconnect", b""));
+        assert_eq!(parse_field(b"disconnect"), field("disconnect", ""));
 
-        assert_eq!(parse_field(b" : foo"), field(" ", b"foo"));
-        assert_eq!(parse_field(b"\xe2\x98\x83: foo"), field("☃", b"foo"));
+        assert_eq!(parse_field(b" : foo"), field(" ", "foo"));
+        assert_eq!(parse_field(b"\xe2\x98\x83: foo"), field("☃", "foo"));
     }
 
-    fn event(typ: &str, data: &[u8], id: &[u8]) -> SSE {
+    fn event(typ: &str, data: &[u8]) -> SSE {
         SSE::Event(Event {
-            data: data.to_vec(),
-            id: id.to_vec(),
+            data: from_utf8(data).unwrap().to_string(),
+            id: None,
```

Comment:
It is not possible to successfully parse an event from non UTF8 data, so it shouldn't be possible to construct one with this helper (hence the `unwrap`).